packages/app-store/salesforce/lib/CrmService.ts

```diff
@@ -472,16 +472,16 @@ export default class SalesforceCRMService implements CRM {
             createdContacts.push(...createdAccountContacts);
           }
         }
+      } else {
```

Comment:
If `appOptions.createNewContactUnderAccount` was true and we were creating the contact, we weren't exiting out of the logic to create the lead record. Thus two records were created at the same time.